services/src/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/links/IslMinPortSpeed.groovy

```diff
@@ -0,0 +1,53 @@
+package org.openkilda.functionaltests.spec.links
+
+import static org.junit.Assume.assumeTrue
+import static org.openkilda.messaging.info.event.IslChangeType.DISCOVERED
+import static org.openkilda.messaging.info.event.IslChangeType.MOVED
+import static org.openkilda.testing.Constants.WAIT_OFFSET
+
+import org.openkilda.functionaltests.BaseSpecification
+import org.openkilda.functionaltests.helpers.Wrappers
+import org.openkilda.testing.model.topology.TopologyDefinition.Isl
+
+import spock.lang.Narrative
+
+@Narrative("""Minimal port speed value is chosen for ISL capacity.
+Sometimes an ISL have different port speed on its edges.
+In that case, we need to set ISL capacity and all bandwidth parameters according to minimal speed value.
+Eg. 10G on one side, and 1G on another side, the ISL should have a 1G capacity.""")
+class IslMinPortSpeed extends BaseSpecification {
+    def "System is able to set min port speed for isl capacity"() {
+        given: "Two ports with different port speed"
+        def isl = topology.islsForActiveSwitches.find { it.getAswitch()?.inPort && it.getAswitch()?.outPort }
+        def port = northbound.getPort(isl.srcSwitch.dpId, isl.srcPort)
+        assumeTrue("Wasn't able to find required a-switch links", isl as boolean)
+
+        def notConnectedIsls = topology.notConnectedIsls
+        def newDst = notConnectedIsls.find {
+            northbound.getPort(it.srcSwitch.dpId, it.srcPort).maxSpeed != port.maxSpeed
```

Comment:
I assume this test is hardware-only?

Reply:
You are right, now it is working on hardware environment only

I guess we can leave it as is(Current implementation doesn't break local environment)
or Should I add _requireProfile_ ?

Reply:
yeah, I guess we can leave as is